gnovm/pkg/doc/json_doc.go

```diff
@@ -113,32 +143,126 @@ func (d *Documentable) WriteJSONDocumentation() (*JSONDocumentation, error) {
 	for _, fun := range pkg.Funcs {
 		jsonDoc.Funcs = append(jsonDoc.Funcs, &JSONFunc{
 			Name:      fun.Name,
-			Signature: mustFormatNode(d.pkgData.fset, fun.Decl),
+			Crossing:  isCrossing(fun),
+			Signature: mustFormatNode(d.pkgData.fset, fun.Decl, opt.Source, file),
 			Doc:       string(pkg.Markdown(fun.Doc)),
 			Params:    d.extractJSONFields(fun.Decl.Type.Params),
 			Results:   d.extractJSONFields(fun.Decl.Type.Results),
 		})
 	}
 
 	for _, typ := range pkg.Types {
+		// Set typeSpec to the ast.TypeSpec within the declaration that defines the symbol.
+		var typeSpec *ast.TypeSpec
+		for _, spec := range typ.Decl.Specs {
+			tSpec := spec.(*ast.TypeSpec) // Must succeed
+			if typ.Name == tSpec.Name.Name {
+				typeSpec = tSpec
+				break
+			}
+		}
+		if typeSpec == nil || typeSpec.Type == nil {
+			// We don't expect this
+			continue
+		}
+		typeExpr := typeSpec.Type
+
+		// De-parenthesize
+		for {
+			if t, ok := typeExpr.(*ast.ParenExpr); ok {
+				typeExpr = t.X
+			} else {
+				break
+			}
+		}
+
+		kind := ""
+		var methods []*JSONFunc
+		var fields []*JSONField
+
+		switch t := typeExpr.(type) {
+		case *ast.StructType:
+			kind = structKind
+			// TODO: Anonymous fields.
+			fields = d.extractJSONFields(t.Fields)
```

Comment:
Are they not handled?

If we don't handle them yet, we can do it next time together with embedded interfaces.

Reply:
"TODO: Anonymous fields" is in the current code. Let's fix it next time.
https://github.com/gnolang/gno/blob/d5a4e89857b6eb4502c8fad635a82b3a25925d81/gnovm/pkg/doc/print.go#L896C13-L896C17